src/Aspire.Hosting/Orchestrator/ApplicationOrchestrator.cs

```diff
@@ -179,7 +179,10 @@ private async Task ProcessUrls(IResource resource, CancellationToken cancellatio
         // Run the URL callbacks
         if (resource.TryGetAnnotationsOfType<ResourceUrlsCallbackAnnotation>(out var callbacks))
         {
-            var urlsCallbackContext = new ResourceUrlsCallbackContext(new(DistributedApplicationOperation.Run), resource, urls, cancellationToken)
+            var executionContext = new DistributedApplicationExecutionContext(
```

Comment:
While you're here, this context be injected in the ctor. The DistributedApplicationExecutionContext. Then there's no need to set it up manually.

Reply:
Keep the IServiceProvider on the ApplicationOrchestrator constructor I presume and just add the DistributedApplicationExecutionContext?

Reply:
Yep

Reply:
Will do it now